Source/WebKit/UIProcess/Inspector/win/InspectorResourceLocation.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2022 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "InspectorResourceLocation.h"
+
+#if ENABLE(REMOTE_INSPECTOR)
+
+#include <wtf/FileSystem.h>
+
+namespace WebKit {
+
+static String bundlePath()
+{
+    wchar_t buffer[MAX_PATH];
+    DWORD length = ::GetModuleFileNameW(nullptr, buffer, WTF_ARRAY_LENGTH(buffer));
+
+    String path(buffer, length);
+    return FileSystem::parentPath(path);
+}
+
+namespace InspectorResource {
+
+String backendCommandsPath()
+{
+    return FileSystem::pathByAppendingComponents(bundlePath(), { "WebKit.Resources"_s, "WebInspectorUI"_s, "Protocol"_s, "InspectorBackendCommands.js"_s });
+}
+
+String uiPath(StringView resource)
```

Comment:
I assume the code is tested, so these questions must already have sensible answers, but I am wondering: Is the `resource` argument allowed to have separators in it? If so, are they `/` characters or `\` characters on Windows? Does the `pathByAppendingComponents` implementation support arguments with separators in them?

Reply:
File removed so resolving